src/main/java/com/gluonhq/richtext/model/PieceTable.java

```diff
@@ -165,15 +165,18 @@ static boolean inRange( int index, int start, int length ) {
 
     private final String text;
     private Piece newPiece;
+    private boolean execSuccess = false;
 
     AppendCmd(String text) {
         this.text = Objects.requireNonNull(text);
     }
 
     @Override
     protected void doUndo(PieceTable pt) {
-        pt.pieces.remove(newPiece);
-        pt.fire( new TextBuffer.DeleteEvent(pt.getTextLength()-text.length(), text.length() ));
+        if (execSuccess) {
+            pt.pieces.remove(newPiece);
```

Comment:
I've noticed that after appending and undoing the  change, pt.textLength is not updated (has the value of the appended text). Do we need to call?
```
pt.textLength = pt.getTextLength() - text.length();
```

Reply:
Good catch. Since append command is just removing one piece and not using `insert` or `delete` methods - I missed that. Will take care of this.
Also I think we have to convert textLength to a property in both model and viewModel.

Reply:
Done